src/prefect/states.py

```diff
@@ -614,7 +615,13 @@ def Completed(cls: Type[State[R]] = State, **kwargs: Any) -> State[R]:
     Returns:
         State: a Completed state
     """
-    return cls(type=StateType.COMPLETED, **kwargs)
+    state_details = StateDetails.model_validate(kwargs.pop("state_details", {}))
```

Comment:
I've connected this to the `Completed` state, because I believe that we require a `Completed` state when passing futures as parameters.  If there are other states that should be covered here, let me know and we can move this into like a `__post_init__`/validator for `StateDetails` so it's present on all states.

Reply:
I think we may need this on the `Failed` state also because you can use the `allow_failure` annotation to receive futures in downstream tasks and flows with a failed state. Is there any downside to adding this to all the state types?

Reply:
@bunchesofdonald and I were waffling on that yesterday.  The only downside I could think of was a little more data stored with each state.  You sold me, I'll move this into a validator that can attach this to all state details

Reply:
Oops, I can't do this universally in a `model_validator` on `StateDetails` because it could mean that we accidentally pick up a trace/span any random time we deserialize this from the API.  I'll be more targeted and add it to `Failed` for now

Reply:
Actually I went another way with it with a small helper in `states.py`, let me know what you think @desertaxle